### ISSUE TITLE
Superset i18n: Fix language settings in echarts

### ISSUE THREAD

**Issue by pomegranited** [openedx/openedx-aspects#300]
### Description

When you switch language in Superset, some charts still have untranslated buttons.
For example Pie chart has 'All' and 'Inv' buttons, that are always on English.

### Supporting information

https://github.com/apache/superset/issues/30131

### Proposed solution

Pass the current `locale` object to echart init, see comments from here https://github.com/apache/superset/issues/30131#issuecomment-2326551298

### Aspects specifics

- [ ] Ensure the solution is backported to the Superset version used by Aspects. ❌ see [comment](https://github.com/openedx/openedx-aspects/issues/300#issuecomment-2767884629)

**Comment by bmtcril** [comment]
Just noting that this is an upstream Superset enhancement, but is an improvement we want for Aspects.

**Comment by pomegranited** [comment]
https://github.com/apache/superset/pull/31751 addresses this issue, submitted by another Superset community member.

**Comment by jpchev** [comment]
> [apache/superset#31751](https://github.com/apache/superset/pull/31751) addresses this issue, submitted by another Superset community member.

thank you for considering me a Superset community member, it's a honor 😄 
Do you know whether that PR is going to be merged?

Thank you

**Comment by pomegranited** [comment]
@jpchev  Evan is on it :)

**Comment by bmtcril** [comment]
@pomegranited I'm not sure what all that entails, or what version of Superset will be out for Teak but we'd definitely like this whenever it's feasible to get it on a supported release!

**Comment by pomegranited** [comment]
@bmtcril I can think of 2 ways to handle this:

1. Create our own superset image builds using our own fork of the Superset version Aspects uses, or
2. Apply our code patches onto the superset-maintained image build from our own fork during the Tutor build.

1 requires more overhead, but 2 is somewhat riskier/more complex.. do you have a preference? I'm hoping to get https://github.com/openedx/openedx-aspects/issues/299 merged soon too, so would backport both together -- they're both really tiny.

Also, do you know roughly when Aspects will settle on a Superset release version for Teak?

**Comment by bmtcril** [comment]
I don't think we really have the bandwidth to maintain either of these off hand. I was hoping it squeaked into 4.1.2, but it looks like it didn't. We may have to wait on this one, unfortunately, unless you think that option 2 is pretty low risk.

We just released Aspects 2.0 with the breaking changes, but are hoping to get the in-context metrics work in for Teak but none of that has landed yet so I'm not sure what version it will be. I'm doing final testing on the beta server before asking that 2.0 go onto the Teak sandbox, but I don't think that will be the final version.

**Comment by pomegranited** [comment]
@bmtcril Ya, I was really hoping to get it all into 4.1.2 too, but no luck :(

> We may have to wait on this one, unfortunately, unless you think that option 2 is pretty low risk.

I think waiting is the most efficient use of our time and the least risk, so ya, will have to wait for Ulma.
